propagation.go

```diff
@@ -0,0 +1,101 @@
+package opentracing
+
+///////////////////////////////////////////////////////////////////////////////
+// CORE PROPAGATION INTERFACES:
+///////////////////////////////////////////////////////////////////////////////
+
+// Injector is responsible for injecting Span instances in a manner suitable
+// for propagation via a format-specific "carrier" object. Typically the
+// injection will take place across an RPC boundary, but message queues and
+// other IPC mechanisms are also reasonable places to use a Injector.
+//
+// See Extractor and Tracer.Injector.
+type Injector interface {
+	InjectSpan(span Span, carrier interface{}) error
+}
+
+// Extractor is responsible for extracting Span instances from an
+// format-specific "carrier" object. Typically the extraction will take place
+// on the server side of an RPC boundary, but message queues and other IPC
+// mechanisms are also reasonable places to use a Extractor.
+//
+// See Injector and Tracer.Extractor.
+type Extractor interface {
+	JoinTrace(operationName string, carrier interface{}) (Span, error)
```

Comment:
agreed... same with InjectSpan. I'll do this stuff when I'm back at my computer in an hour.